src/Repository.ts

```diff
@@ -301,16 +305,28 @@ export class MongoRepository<DOC, DTO = DOC> {
    * @param {string} type any of the valid types, PRE_KEY POST_KEY
    * @param {RepoOperation[]} fns any of the valid functions: update, updateOne, save, create, find, findOne, findMany
    * @param {*} newDocument The document to apply functions to
-   * @param {*} oldDocument The original document before changes were applied
+   * @param originalDocument The original document before changes were applied
+   * @param opts Options for event
    * @returns {Promise<any>}
    * @memberof MongoRepository
    */
   protected async invokeEvents(
     type: string,
     fns: RepoOperation[],
     newDocument: any,
-    originalDocument?: any
+    originalDocument?: any,
+    opts?: EventOptions
   ): Promise<any> {
+    // local options override global collection options
+    if (opts?.noClone === false || (opts?.noClone === undefined && this.options.eventOpts?.noClone !== true)) {
```

Comment:
Does this turn on cloning by default? If so, then maybe this should be a major semver bump?

Reply:
it does 👍

Reply:
Fixed.

Reply:
😇